react/src/pages/admin/UserList.tsx

```diff
@@ -1,139 +1,152 @@
+                dispatch({
+                    type: "set",
+                    payload: data,
+                })
+            ); // No safety check on JSON structure
     }, []);
 
-    const classes = useStyles();
-
     return (
-        <main className={classes.root}>
+        <>
             <CreateUserModal
                 id="create-modal"
-                open={addingUser}
-                onClose={() => setAddingUser(false)}
-                onSuccess={addUserSuccess}
-            />
-            <ConfirmModal
-                id="confirm-modal-update"
-                color="primary"
-                open={!!updatingUser}
-                onClose={() => setUpdatingUser(null)}
-                onConfirm={updateUser}
-                title="Confirm updating user"
-            >
-                Update {updatingUser && updatingUser.username} and maybe overwrite password?
-            </ConfirmModal>
-            <ConfirmModal
-                id="confirm-modal-delete"
-                color="secondary"
-                open={!!deletingUser}
-                onClose={() => setDeletingUser(null)}
-                onConfirm={deleteUser}
-                title="Delete user"
-            >
-                Really delete {deletingUser && deletingUser.username}?
-            </ConfirmModal>
-            <Snackbar
-                anchorOrigin={{ vertical: "bottom", horizontal: "center" }}
-                open={!!message}
-                onClose={() => setMessage("")}
-                message={message}
-                color={messageColor}
+                open={openNewUser}
+                onClose={() => setOpenNewUser(false)}
+                onSuccess={user => {
+                    dispatch({ type: "add", payload: user });
+                    setOpenNewUser(false);
+                    enqueueSnackbar(`New user ${user.username} created successfully`);
+                }}
             />
-            <Button
-                variant="contained"
-                color="primary"
-                className={classes.addButton}
-                onClick={() => setAddingUser(true)}
-            >
-                <AddIcon />
-                Add new
-            </Button>
-            <Grid container spacing={2}>
-                {userList.map(user => (
-                    <Grid item xs={12} md={6}>
+            <Toolbar component={Paper} className={classes.toolbar}>
+                <Typography variant="h6">Users</Typography>
+                <Box className={classes.grow} />
+                <Tooltip title="Add new user">
+                    <IconButton onClick={() => setOpenNewUser(true)}>
+                        <PersonAdd />
+                    </IconButton>
+                </Tooltip>
+            </Toolbar>
+            <Grid container spacing={1} alignItems="flex-start">
+                {users.map(user => (
+                    <>
                         <UserRow
-                            key={user.username}
```

Comment:
The key should remain. Also the only component that's mapped each time is UserRow so there's no need for the surrounding fragment `<></>`.